transports/ssl/src/main/java/org/mule/transport/ssl/revocation/CrlFile.java

```diff
@@ -49,22 +50,14 @@ public void setPath(String path)
     }
 
     @Override
-    public ManagerFactoryParameters configFor(KeyStore trustStore)
+    public ManagerFactoryParameters configFor(KeyStore trustStore, Set<TrustAnchor> defaultTrustAnchors)
     {
+        assertNotNull(trustStore, "tls:crl-file requires the 'path' attribute");
+        assertNotNull(trustStore, "tls:crl-file requires a trust store");
+
         try
         {
-            // When creating build parameters we must manually trust each certificate (which is automatic otherwise)
-            Enumeration<String> aliases = trustStore.aliases();
-            HashSet<TrustAnchor> trustAnchors = new HashSet<>();
-            while (aliases.hasMoreElements())
-            {
-                String alias = aliases.nextElement();
-                if (trustStore.isCertificateEntry(alias))
-                {
-                    trustAnchors.add(new TrustAnchor((X509Certificate) trustStore.getCertificate(alias), null));
-                }
-            }
-
+            Set<TrustAnchor> trustAnchors = getTrustAnchorsFromKeyStore(trustStore);
```

Comment:
Why is this defined in `TlsConfiguration`? I've only seen it used here.

Reply:
It could be useful for future revocation code. But if you want I can move it there.

Reply:
Yes, move it to where it's used